features_and_tools/firely-server-ingest.rst

```diff
@@ -301,6 +301,25 @@ The cached files can be found in the following locations:
   * Windows: ``%APPDATA%\.fhir_packages``
   * Linux/MacOS: ``$XDG_CONFIG_HOME/.fhir_packages`` if the environment variable ``XDG_CONFIG_HOME`` is defined  otherwise ``$HOME/.config/.fhir_packages``
 
+.. _tool_fsi_recovery:
+
+Recovery Journal
+----------------
+
+If a transient error occurs while ingestion is running or the FSI instance gets interrupted, the *recovery journal* feature allows recovery from such a situation. To enable it, use the ``--useRecoveryJournal <recoveryJournalDirectory>`` option in the CLI or set field ``recoveryJournalDirectory`` in the ``appsettings.instance.config``. 
+
+When enabled, the process runs as follows:
+
+#. Upon the first ingestion attempt, FSI will take a snapshot of all the files in the specified source directory and save that snapshot to the ``<recoveryJournalDirectory>``.
+#. Then the data ingestion will start. Information about every successfully ingested resource also gets added to the journal.
+
+If the ingestion procedure gets interrupted at any point, or some of the resources do not get ingested because of a transient error (e.g. network connection to the target DB is temporarily down), the ingestion process can be restarted by running the application with the same parameters. The application will skip all the previously ingested resources based on the journal.
+
+.. note::
+  
+  - Note that the recovery journal directory must be empty before performing the initial ingestion attempt for a given set of files. 
```

Comment:
@eugene-blokhin, I have not tested this yet, but upon reading this makes me wonder: Will Firely Server throw an error if a new ingestion attempt is done without removing this file?

Reply:
It indeed. That's the next bullet in the list :)

Shall I rephrase it?